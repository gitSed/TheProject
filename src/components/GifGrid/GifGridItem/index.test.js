import React from 'react';

/* Third-Party */
import { shallow } from 'enzyme';

/* Own */
import GifGridItem from '.';

describe('Testing <GifGridItem />', () => {
  const title = 'Un titulo';
  const url = 'https://localhost.com';
  const wrapper = shallow(<GifGridItem title={title} url={url} />);

  test('Render', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('Paragraph render', () => {
    const p = wrapper.find('p');

    expect(p.text().trim()).toBe(title);
  });

  test('Image render', () => {
    const img = wrapper.find('img');

    expect(img.prop('src')).toBe(url);
    expect(img.prop('alt')).toBe(title);
  });

  test('ClassName animate__fadeIn', () => {
    const div = wrapper.find('div');

    expect(div.hasClass('animate__fadeIn')).toBe(true);
  });
});
