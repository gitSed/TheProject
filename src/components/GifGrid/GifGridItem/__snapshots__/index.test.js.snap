// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <GifGridItem /> Render 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Un titulo"
    src="https://localhost.com"
  />
  <p>
    Un titulo
  </p>
</div>
`;
